packages/TorneloScoresheet/src/pages/TablePairing/TablePairing.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import { useTablePairingState } from '../../context/AppModeStateContext';
+import { ChessGameInfo } from '../../types/ChessGameInfo';
+import { styles } from './style';
+import { Text, TouchableOpacity, View } from 'react-native';
+import Sheet from '../../components/Sheet/Sheet';
+import { chessEngine } from '../../chessEngine/chessEngineInterface';
+import PrimaryButton from '../../components/PrimaryButton/PrimaryButton';
+import { PieceType } from '../../types/ChessMove';
+import { PlayerColour } from '../../types/ChessGameInfo';
+import PieceAsset from '../../components/PieceAsset/PieceAsset';
+import PrimaryText, {
+  FontWeight,
+} from '../../components/PrimaryText/PrimaryText';
+
+const TablePairing: React.FC = () => {
+  const tablePairingState = useTablePairingState();
+  const tablePairingMode = tablePairingState?.[0];
+  const [showSheet, setShowSheet] = useState(false);
+  const [selectedPlayer, setSelectedPlayer] = useState(0);
+  const goToRecording = tablePairingState?.[1].goToRecording;
+
+  //TO DO: LIMIT LENGTH OF INPUT NAME
+  const setPlayer = (playerNumber: number) => {
+    setSelectedPlayer(playerNumber);
+  };
+
+  const handleConfirm = () => {
+    if (!tablePairingMode || !goToRecording) {
+      return;
+    }
+
+    const [board, fen] = chessEngine.startGame();
+    goToRecording(
+      tablePairingMode.pairing,
+      [{ moveNo: 1, whitePly: { startingFen: fen } }],
+      board,
+    );
+  };
```

Comment:
this type of logic is linked to transitioning between states and as such does not belong in the view, the code to start the game and build the movehistory should be in your useTablePairingState() hook and you can access the pairing from inside the hook too, and you should just call goToRecording() with no arguments from here.